modules/aws-eks-self-managed-node-groups/locals.tf

```diff
@@ -49,16 +51,20 @@ locals {
     var.self_managed_ng
   )
 
+  # WINDOWS CONFIGURATION
+  compatible_with_windows_server_2022 = var.context.cluster_version >= "1.22" ? true : false 
```

Comment:
1. Can users create clusters with 2019 AMI on 1.22/1.23?
2. Can users create clusters with 2022 AMI on 1.19-1.22?

Also, this is not a valid comparison `var.context.cluster_version >= "1.22"`, this is not a numerical comparison

I think what we want to do is add a variable for the `windows_server_version` and have it default to `2019` but users have the ability to pass in `2022` and then we can get rid of the other changes

Reply:
We are looking to change the default to 2022 instead of 2019, but do not exist AMI for Windows Server 2022 that are compatible with cluster below 1.22, so the default must be 2022 for all clusters that are in 1.22 or above and 2019 for cluster that are below.

Reply:
Is it possible to describe that on the new version disclaimer and make this new version default? This behavior is intentional.